src/reducers.js

```diff
@@ -40,28 +40,30 @@ export function Reducer(prevState = {}, action) {
     case 'update_metapaths':
       if (action.payload.metapaths !== undefined) {
         newState.metapaths = action.payload.metapaths;
-        transferObjectProps(
-          prevState.metapaths,
-          newState.metapaths,
-          ['id'],
-          ['checked', 'highlighted']
-        );
+        if (action.transferState === true) {
+          transferObjectProps(
+            prevState.metapaths,
+            newState.metapaths,
```

Comment:
seems like this also modifies `prevState.metapaths`?

Reply:
It shouldn't. That function only modifies the newState, or the `newArray` prop of the function.

Reply:
oh ok, I was confused because it seems that `prevState.metapaths` and `newState.metapaths` have a reference to the same object.

Reply:
In my testing, I didn't notice that they were referencing the same object. I thought `const newState = { ...prevState };` would necessarily make a copy. I will re-test this though; this could very well be a glaring error.

Reply:
As best I can tell, `const newState = { ...prevState }` behaves the same as the `Object.assign({}, prevState)` approach [suggested by redux](https://redux.js.org/basics/reducers). 

From my research ([1](https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript),[2](https://stackoverflow.com/questions/32925460/object-spread-vs-object-assign)), it seems like both approaches generate a "shallow copy", with a "deep copy" requiring significantly more effort.

Reply:
hmm I think that modifying `prevState` might lead to [problems later](https://redux.js.org/recipes/structuring-reducers/prerequisite-concepts#note-on-immutability-side-effects-and-mutation) with other redux features. [This](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#common-mistake-2-only-making-a-shallow-copy-of-one-level) has suggestions on how to create the new objects.

Reply:
I don't think I ever planned to use the other features, except for maybe the time travel debugging at some point. I wonder why they show the `Object.assign` approach in the reducer example when it has the same shortcomings of the `{ ... ` approach they say is a mistake later...

To avoid writing a huge custom function to deep copy, for now I think I will use the standard `JSON.parse(JSON.stringify(obj))`. I've seen comments on SO that although it's slow, it's still as fast or faster as the other techniques of manually deep cloning by looping and such. Thoughts?